Sources/MapboxCoreNavigation/Router.swift

```diff
@@ -397,7 +397,8 @@ extension InternalRouter where Self: Router {
         routeTask?.cancel()
         let options = progress.reroutingOptions(from: origin)
         
-        if isRerouting {
+        // https://github.com/mapbox/mapbox-navigation-ios/issues/3966
+        if isRerouting && options.profileIdentifier != .walking && options.profileIdentifier != .cycling {
```

Comment:
I went with a negation instead of checking for the `automobile` and `automobileAvoidingTraffic` profiles because other undocumented profiles may support this query parameter. I’m open to moving this conditional to MapboxDirections, where `initialManeuverAvoidanceRadius` is converted into a `URLQueryItem`, but it seems like we’ve been shying away from hard-coding per-profile support for various parameters in that library, lest the Directions API expand support without us knowing.

/cc @LukasPaczos

Reply:
A problem with negation is that if there's an undocumented profile that does not support this query parameter, the whole request will fail, won't it? Not sure what's the lesser of the two - not supporting a query parameter until SDK is updated, or risking a failing request. I think I'd rather opt for the former, especially since this is a parameter that SDK sets automatically under-the-hood for each reroute request and developer cannot exclude it if they use a profile that does not support this parameter.

Reply:
Good point. I don’t know of an undocumented profile that lacks support for this parameter, but it’s entirely possible that one may arrive on scene without us knowing on the client side.

I don’t think we can confidently add parameters under the hood like this without giving the developer the opportunity to intervene. #3464 was all about giving the developer this flexibility, but the developer has to implement a full-fledged custom routing provider to take advantage of it. I think it would make sense for… something… to expose a customization hook for the options used when rerouting.

In the meantime, I’ll reverse the negation to fix the immediate regression in the walking and cycling profiles, which are documented.

Reply:
>  giving the developer the opportunity to intervene

On the Android side we did recently add a `RerouteOptionsAdapter` which does just that - https://github.com/mapbox/mapbox-navigation-android/pull/5573. The same would have to be added to all native controllers that execute requests on their own for the same precautions though. That said, I'd still feel more confident with solid "backwards compatibility" than "forward compatibility" (if we can call the different paradigms this way).

Reply:
Tracking the forward-compatibility story in #4032.